### DISCUSSION TITLE
Reached heap limit Allocation failed - JavaScript heap out of memory

### DISCUSSION THREAD

**Discussion by dasshiba** [nodejs/help#4747] [Q&A]
### Version

16.14.2

### Platform

Angular CLI: 14.2.11 

### What steps will reproduce the bug?

We are migrating the cms product from angular v4 to angular v14. While trying to run the angular v14 application locally (command ng serve), we are getting the following error - "FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory".

We have tried to add more space as well by executing the command "node -max_old_space_size=8192./node_modules/@angular/cli/bin/ng serve". But still, the issue is not resolved.

And We can't take Heap Snapshot from our application , we noticed that when we run ng build command the node.exe process consumes memory up to 11 GB .

### What do you see instead?

We are migrating the cms product from angular v4 to angular v14. While trying to run the angular v14 application locally (command ng serve), we are getting the following error - "FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory".

We have tried to add more space as well by executing the command "node -max_old_space_size=8192./node_modules/@angular/cli/bin/ng serve". But still, the issue is not resolved.

The screenshot of the error is attached here for reference. It would be helpful if you can provide any suggestions to resolve this issue.
![ScreenShot_3](https://github.com/nodejs/node/assets/132643621/dd9a51f0-ff9e-489a-bf12-aba99cdc72c7)

And We can't take Heap Snapshot from our application , we noticed that when we run ng build command the node.exe process consumes memory up to 11 GB .

![unnamed](https://github.com/nodejs/node/assets/132643621/652c3688-72e0-4eda-9c8e-27a8413997eb)

### Additional information

**Our Application version details :**

Angular CLI: 14.2.11
Node: 16.14.2
Package Manager: npm 8.5.0
OS: win32 x64

Angular: 14.3.0
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1402.11
@angular-devkit/build-angular   14.2.11
@angular-devkit/core            14.2.11
@angular-devkit/schematics      14.2.11
@angular/cli                    14.2.11
@angular/material               14.2.7
@schematics/angular             14.2.11
rxjs                            7.5.7
typescript                      4.7.4

**Comment by bnoordhuis** [comment]
I've converted this to a discussion because there is no indication it's a node bug (and, ergo, doesn't belong on the bug tracker.)
